src/special/webpack.js

```diff
@@ -39,14 +42,82 @@ function getLoaders(deps, loaders) {
     .value();
 }
 
+function parseWebpack1(filepath, deps) {
+  const module = require(filepath).module || {}; // eslint-disable-line global-require
+  const loaders = getLoaders(deps, module.loaders);
+  const preLoaders = getLoaders(deps, module.preLoaders);
+  const postLoaders = getLoaders(deps, module.postLoaders);
+  return [...loaders, ...preLoaders, ...postLoaders];
+}
+
+function mapRuleLoaders(module) {
+  return module.rules
+    .filter(rule => rule.loaders)
+    .map(rule => rule.loaders
+      .map(loader => ({
+        loader,
+      })));
+}
+
+function mapRuleUse(module) {
+  return module.rules
+    // filter use or loader (loader is a shortcut to use)
+    .filter(rule => rule.use || rule.loader)
+    .map((rule) => {
+      const key = rule.use ? 'use' : 'loader';
+
+      // map simple strings as { loader: string }
+      if (typeof rule[key] === 'string') {
+        return {
+          loader: rule[key],
+        };
+      }
+
+      // map objects without changing anything
+      if (typeof rule[key] === 'object' && !lodash.isArray(rule[key])) {
+        return rule[key];
+      }
+
+      // if it's an array, apply the two rules above for each element
+      if (lodash.isArray(rule[key])) {
+        return rule[key].map((value) => {
+          if (typeof value === 'object') {
+            return value;
+          }
+
+          return {
+            loader: value,
+          };
+        });
+      }
+
+      throw new Error('Unrecognised rule.loader format');
+    });
+}
+
+function parseWebpack2(filepath, deps) {
+  try {
+    const module = require(filepath).module || {}; // eslint-disable-line global-require
+    if (!module.rules) {
+      return [];
+    }
+
+    const mappedLoaders = [];
+    mappedLoaders.push(...mapRuleLoaders(module));
+    mappedLoaders.push(...mapRuleUse(module));
+
+    return getLoaders(deps, lodash.flatten(mappedLoaders));
```

Comment:
Is this `lodash.flatten` necessary? I think you have good structure at this moment.

Reply:
It is - `mapRuleUse` can return an array of arrays because `use` or `loader` can be an array.  Eg: 
```js
module: {
      rules: [
        { test: /\.css$/, loader: [{ loader: 'style-loader' }] },
      ],
    },
```

Rather than test the type of each `loader` or `use` property it's easier to just flatten the whole lot.

Reply:
I've made the rest of the changes - thank you for your excellent feedback! I'll wait until we've worked out what's going on here ^ before resubmitting the PR.